tutorials/arvo/arvo.md

```diff
@@ -0,0 +1,519 @@
+
+ + The standard library, `zuse`, is installed.
+ + Entropy is added
+ + Identity is added
+ + Metamorph into the next stage of Arvo
+
+Once the larval stage has passed its functionality will never be used again.
+
+## The state
+
+As we follow functional programming paradigms, the state of Arvo is considered
+to be the entire Arvo kernel core currently in operation (whether it be the larval
+stage or adult stage). Thus when `+poke` is performed, a new core with the
+updated state it produced, rather than modifying the existing core as would be
+expected to happen in an imperative setting.
+
+Thus besides the battery of the Arvo core, we have the payload which is as
+follows.
+
+```hoon
+::  persistent arvo state
+::
+=/  pit=vase  !>(..is)                                  ::
+=/  vil=vile  (viol p.pit)                              ::  cached reflexives
+=|  $:  lac=_&                                          ::  laconic bit
+        eny=@                                           ::  entropy
+        our=ship                                        ::  identity
+        bud=vase                                        ::  %zuse
+        vanes=(list [label=@tas =vane])                 ::  modules
+    ==                                                  ::
+```
+
+```hoon
+=/  pit=vase  !>(..is)                                  ::
+```
+This `vase` is part of the state but does not get directly migrated when `+poke`
+is called. `!>(..is)` consists of the code in `arvo.hoon` written above this core contained in
+a `vase`. Thus this part of the state changes only when that code changes in an
+update.
+
+```hoon
+=/  vil=vile  (viol p.pit)                              ::  cached reflexives
+
+```
+This is a cache of specific types that are of fundamental importance to Arvo -
+namely `type`s, `duct`s, `path`s, and `vase`s. This is kept because it is
+unnecessarily wasteful to recompile these fundamental types on a regular basis.
+Again, this part of the state is never updated directly by `+poke`.
+
+```hoon
+=|  $:  lac=_&                                          ::  laconic bit
+        eny=@                                           ::  entropy
+        our=ship                                        ::  identity
+        bud=vase                                        ::  %zuse
+        vanes=(list [label=@tas =vane])                 ::  modules
+    ==
+```
+This is where the real state of the Arvo kernel is kept. `lac` detemines whether
+Arvo's output is verbose, which can be set using the `|verb` command in the
+dojo. `eny` is the current entropy. `our` is the ship, which is permanently
+frozen during the larval stage. `bud` is the standard library. Lastly, `vanes`
+is of course the list of vanes, which have their own internal states. As Hoon is
+homoiconic, code is data, so this is does secrely have everything the vanes are
+doing encoded within, including all user apps.
```

Comment:
This sentence doesn't make sense.  Not sure what "secrely" was supposed to be, but I also don't think this is connected to homoiconicity.

Reply:
alright I just removed the sentence